### DISCUSSION TITLE
How to make eto FontFamily to use embeded fonts?

### DISCUSSION THREAD

**Discussion by WileyNg** [picoe/Eto#2325] [General]
Hi @cwensley,

Yet another question about eto. I use the following to embed fonts in my rhino Plugin.

```             C#
            string[] fileInfos = Assembly.GetManifestResourceNames().Where(x=> x.Split('.').Last()=="ttf").ToArray();

            foreach (string fileinfo in fileInfos)
            { 
            Stream fontStream = this.GetType().Assembly.GetManifestResourceStream(fileinfo);
            byte[] fontdata = new byte[fontStream.Length];
            fontStream.Read(fontdata, 0, fontdata.Length);
            fontStream.Close();
            unsafe
            {
                fixed (byte* pFontData = fontdata)
                    {
                        FontCollection.AddMemoryFont((System.IntPtr)pFontData, fontdata.Length);
                    }
            }
```

However, I don't see eto has any way to use System.Drawing.FontFamily class? 
there is no relevant constructor in the Eto.Drawing.FontFamily to take that in.

any ideas?

**Comment by cwensley** [comment]
Hey @WileyNg, try `FontFamily.FromStreams()`. For a single typeface you can use the `FontTypeface` constructors.

**Reply by WileyNg** [reply]
Thanks so much for your prompt response!! @cwensley

Unfortunately I couldn't find the answer.
there is no FontTypeface constructor that can take in a System.Drawing.FontFamily class. Do you mind posting a quick example?
There is also no method called .FromStreams() under FontFamily :(

**Reply by cwensley** [reply]
No, I mean use those API's to load your font(s) from stream or file then use it directly. You don't need to use System.Drawing at all.

**Reply by WileyNg** [reply]
but how would a Label in eto be able to use those fonts? Label only takes in eto fonts? 
really appreciate your help!

**Reply by cwensley** [reply]
```cs
var fontFamily = Eto.Drawing.FontFamily.FromStreams(myStreams);
var label = new Eto.Forms.Label
{
  Text = "Hello",
  Font = new Eto.Drawing.Font(fontFamily, 12)
};

// or:
var fontTypeface = new Eto.Drawing.FontTypeface(singleFontStreamOrFileName);
var label2 = new Eto.Forms.Label
{
  Text = "Hello again",
  Font = new Font(fontTypeface, 12)
};
```

**Reply by WileyNg** [reply]
![image](https://user-images.githubusercontent.com/5993201/192984579-6fe02be3-b8e2-4a04-a1ad-fe3003c5ff9a.png)
![image](https://user-images.githubusercontent.com/5993201/192984065-a72eabf2-ee3c-4f46-a207-f2559eddcb9d.png)

@cwensley thanks so much for helping.
But I just couldn't find FromStreams method under FontFamily in the Eto assembly reference.
The FontTypeface also has only one constructor available.

**Reply by WileyNg** [reply]
Hi @cwensley 
I updated the eto version and found the FromStream Method! Thanks!

**Reply by cwensley** [reply]
Hey @WileyNg, awesome.  Yes, they are relatively newer APIs from PR #2174, only in 2.7.0 or later.
